tests/integration_tests/test_time_series_integration.py

```diff
@@ -0,0 +1,186 @@
+    def setUp(self):
+        self.nwbfile = pynwb.NWBFile(
+            session_description="Testing inspector.",
+            identifier=str(uuid4()),
+            session_start_time=datetime.now().astimezone(),
+        )
+
+    def test_check_regular_timestamps(self):
+        time_series = pynwb.TimeSeries(
+            name="test_time_series",
+            unit="test_units",
+            data=np.zeros(shape=3),
+            timestamps=[1.2, 3.2, 5.2],
+        )
+        self.nwbfile.add_acquisition(time_series)
+        self.assertEqual(
+            first=check_regular_timestamps(
+                time_series=self.nwbfile.acquisition["test_time_series"]
+            ),
+            second=(
+                "TimeSeries appears to have a constant sampling rate. "
+                f"Consider specifying starting_time={time_series.timestamps[0]} "
+                f"and rate={time_series.timestamps[1] - time_series.timestamps[0]} instead of timestamps."
+            ),
+        )
+
+    def test_check_data_orientation(self):
+        self.nwbfile.add_acquisition(
+            pynwb.TimeSeries(
+                name="test_time_series",
+                unit="test_units",
+                data=np.zeros(shape=(2, 100)),
+                rate=1.0,
+            )
+        )
+        self.assertEqual(
+            first=check_data_orientation(
+                time_series=self.nwbfile.acquisition["test_time_series"]
+            ),
+            second=(
+                "Data orientation may be in the wrong orientation. "
+                "Time should be in the first dimension, and is usually the longest dimension. "
+                "Here, another dimension is longer. "
+            ),
+        )
+
+    def test_check_timestamps_match_first_dimension_1d(self):
+        self.nwbfile.add_acquisition(
+            pynwb.TimeSeries(
+                name="test_time_series",
+                unit="test_units",
+                data=np.zeros(shape=4),
+                timestamps=[1.0, 2.0, 3.0],
+            )
+        )
+        self.assertEqual(
+            first=check_timestamps_match_first_dimension(
+                time_series=self.nwbfile.acquisition["test_time_series"]
+            ),
+            second="The length of the first dimension of data does not match the length of timestamps.",
+        )
+
+    def test_check_timestamps_match_first_dimension_2d(self):
+        self.nwbfile.add_acquisition(
+            pynwb.TimeSeries(
+                name="test_time_series",
+                unit="test_units",
+                data=np.zeros(shape=(2, 3)),
+                timestamps=[1.0, 2.0, 3.0],
+            )
+        )
+        self.assertEqual(
+            first=check_timestamps_match_first_dimension(
+                time_series=self.nwbfile.acquisition["test_time_series"]
+            ),
+            second="The length of the first dimension of data does not match the length of timestamps.",
+        )
```

Comment:
```suggestion
def test_check_timestamps_match_first_dimension_2d(self):
    timeseries = pynwb.TimeSeries(
        name="test_time_series",
        unit="test_units",
        data=np.zeros(shape=(2, 3)),
        timestamps=[1.0, 2.0, 3.0],
    )
    assert check_timestamps_match_first_dimension(time_series) == "The length of the first dimension of data does not match the length of timestamps.",
```

Reply:
are we missing anything if we do it this way?

Reply:
See comment above; just verifies that adding the `TimeSeries` to the `nwbfile` container doesn't magically (or silently) alter the effect of the logic. This might come up for more complicated checks that depend on the build status (`nwbfile` object or written/read from NWBFile path), thinking of the `h5py.Dataset` checks there.

For this specific test, I wouldn't expect any difference either way but tests exist to defy (or hopefully confirm) such expectations.